elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java

```diff
@@ -5,26 +5,191 @@
  */
 package com.yahoo.elide.core.filter;
 
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.PersistentResource;
 import com.yahoo.elide.core.exceptions.InvalidPredicateException;
+import com.yahoo.elide.utils.coerce.CoerceUtil;
+
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
 /**
  * Operator enum for predicates.
  */
 @RequiredArgsConstructor
-public enum Operator {
-    IN("in", true),
-    NOT("not", true),
-    PREFIX("prefix", true),
-    POSTFIX("postfix", true),
-    INFIX("infix", true),
-    ISNULL("isnull", false),
-    NOTNULL("notnull", false),
-    LT("lt", true),
-    LE("le", true),
-    GT("gt", true),
-    GE("ge", true);
+public enum Operator implements BiFunction<Predicate, EntityDictionary, java.util.function.Predicate> {
+    IN("in", true) {
+        @Override
+        public java.util.function.Predicate getFilterFunction(
+                String field, List<Object> values, EntityDictionary dictionary) {
+            return (entity) -> {
+                Object val = PersistentResource.getValue(entity, field, dictionary);
+                if (val == null) {
+                    return false;
+                }
+                List<Object> coercedValues = values.stream()
```

Comment:
No need to hold this `List` object just to scan it again. We can either use a `Set` (faster set membership lookup) or, probably a better solution, instead of running `collect()` just scan for an `anyMatch`. Something like:

```java
values.stream()
  .map(v -> CoerceUtil.coerce(v, val.getClass()))
  .anyMatch(val::equals)
```